crates/wasmtime/src/profiling.rs

```diff
@@ -188,10 +188,12 @@ fn module_symbols(name: String, compiled: &CompiledModule) -> Option<LibraryInfo
         let loc = compiled.func_loc(defined_idx);
         let func_idx = compiled.module().func_index(defined_idx);
         let mut name = String::new();
-        match compiled.func_name(func_idx) {
-            None => name = format!("wasm_function_{}", defined_idx.as_u32()),
-            Some(func_name) => demangle_function_name(&mut name, func_name).unwrap(),
-        };
+        demangle_function_name_or_index(
+            &mut name,
+            compiled.func_name(func_idx),
+            defined_idx.as_u32() as usize,
+        )
+        .unwrap();
```

Comment:
Could we change the `unwrap` to an `expect` where there is a brief message explaining why we expect this never to fail? Something like

```suggestion
        .expect("only errors if the Writer errors, and string writers never error");
```